jobs/dap-collector-ppa-prod/dap_collector_ppa_prod/main.py

```diff
@@ -273,12 +296,18 @@ def check_time_precision(time_precision_minutes, end_collection_date):
     elif time_precision_minutes % MINUTES_IN_DAY != 0:
         # time precision is a day or longer, but is not a multiple of a day
         return f"Task has time precision that is not an even multiple of a day"
-    elif end_collection_date_seconds % (time_precision_minutes*60) != 0:
+    elif (end_collection_date_seconds + ad_start_offset) % (time_precision_minutes*60) != 0:
         # time precision is a multiple of day, but the end does not align with this task's buckets
         return f"{end_collection_date} does not align with task aggregation buckets"
 
     return None
 
+def find_ad_start_offset(task):
+    if "ad_start_date_iso" not in task:
+        return 0
+    ad_start_date_weekday = datetime.datetime.fromisoformat(task["ad_start_date_iso"]).weekday()
+    days_away_from_thursday = (ad_start_date_weekday - UNIX_EPOCH_WEEKDAY) % 7
+    return days_away_from_thursday*MINUTES_IN_DAY*60
```

Comment:
Ah, so this will work for a 7 day collection period, but we also want to make the code flexible enough to collect on any day interval, with the main intervals being 1 day, 7 day, and 30-31 days. I do not think this code will work for a 30-31 day collection window.

I think one solution for this would be to get the diff in minutes (or seconds) between the `start_date` and `end_collection_date`, then mod it by the time_precision_minutes.

Reply:
yeah i somehow thought that we'll only be dealing with either <=1 or 7 day time_precisions 🤦, i'll rewrite this part